server/src/main/java/org/opensearch/index/mapper/FlatObjectFieldMapper.java

```diff
@@ -63,6 +66,7 @@
 public final class FlatObjectFieldMapper extends DynamicKeyFieldMapper {
 
     public static final String CONTENT_TYPE = "flat_object";
+    public static final String DOC_VALUE_NO_MATCH = new String("");
```

Comment:
Instead of the empty string, could you just make this `new Object()`? The type doesn't matter -- it's just a sentinel object.